### ISSUE TITLE
Cannot instantiate `pl.lit` with a Decimal value

### ISSUE THREAD

**Issue by alexander-beedie** [pola-rs/polars#13341]
### Checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pypi.org/project/polars/) of Polars.

### Reproducible example

```python
from decimal import Decimal as D
import polars as pl

with pl.Config( activate_decimals=True ):
    decimal_zero = pl.lit( D(0) )
```

### Log output

```
Traceback (most recent call last):
  File "<ipython-input-13-ebe78b664c56>", line 1, in <module>
    pl.lit(D(0))
  File "/.../polars/py-polars/polars/functions/lit.py", line 150, in lit
    return wrap_expr(plr.lit(item, allow_object))
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^

TypeError: invalid literal value: '0'
```

### Issue description

We cannot instantiate a decimal literal, irrespective of whether `activate_decimals` is True/False.

### Expected behavior

We can create decimal Series; should also be able to create literals, so we can filter/operate on those Series, eg:
```python
df = pl.DataFrame({
    "dec": [
        D("1.6666666666666666666"), 
        D("2.123456789"),
    ],
})

df.filter( dec=D("1.6666666666666666666") )
```
This filter should succeed; instead we get...
```
# TypeError: invalid literal value: '1.6666666666666666666'
```
Note that if we try and filter with an equivalent float it will fail to match, as the target decimal value cannot be exactly represented (the closest float is given as "1.66666666666666674068...")

### Installed versions

<details>

```
--------Version info---------
Polars:               0.20.3-rc.2
Index type:           UInt32
Platform:             macOS-14.2.1-arm64-arm-64bit
Python:               3.11.6 (main, Oct  2 2023, 13:45:54) [Clang 15.0.0 (clang-1500.0.40.1)]
----Optional dependencies----
adbc_driver_manager:  0.8.0
cloudpickle:          3.0.0
connectorx:           0.3.2
deltalake:            0.14.0
fsspec:               2023.12.2
gevent:               23.9.1
matplotlib:           3.8.2
numpy:                1.26.2
openpyxl:             3.1.2
pandas:               2.1.4
pyarrow:              14.0.2
pydantic:             2.5.3
pyiceberg:            0.5.1
pyxlsb:               1.0.10
sqlalchemy:           2.0.24
xlsx2csv:             0.8.1
xlsxwriter:           3.1.9
```

</details>

**Comment by deanm0000** [comment]
The filter fails even if I add that as a column before the filter

```
### this works (as you say)
df.with_columns(zz=pl.Series([D("1.6666666666666")]))
### this fails
df.with_columns(zz=pl.Series([D("1.6666666666666")])).filter(pl.col('dec')>pl.col('zz'))
PanicException: not implemented
```

**Comment by cgevans** [comment]
There seem to be a few difficulties here:

- `pl.lit` can't handle Python decimals; the code to do that is just missing at the moment.
- `pl.repeat`, *if dtype is not specified*, can't handle the *polars* Decimal dtype. 

Using a pl.Decimal (eg, pl.Series with pl.Decimal dtype) and specifying the dtype for pl.repeat does work:

```python
import polars as pl
from decimal import Decimal
pl.select(pl.repeat(pl.lit(Decimal("1_000.2")), 5)) # fails because of pl.lit
a = pl.Series(values=[Decimal("1_000.2")]).cast(pl.Decimal(scale=12))
pl.select(pl.repeat(pl.lit(a), 5)) # fails because of pl.repeat
pl.select(pl.repeat(pl.lit(a), 5, dtype=pl.Decimal(scale=12))) # works
```

**Comment by cgevans** [comment]
There are two problems here:

- `pl.lit` doesn't support Python Decimals (this seems relatively easy to fix, and I have something that seems to work).
- Equality and comparisons aren't implemented for (Polars) Decimals (#12118).
